crates/sui-benchmark/src/workloads/payload.rs

```diff
@@ -11,7 +11,7 @@ use crate::workloads::workload::WorkloadType;
 
 pub trait Payload: Send + Sync + std::fmt::Debug {
     fn make_new_payload(self: Box<Self>, effects: &ExecutionEffects) -> Box<dyn Payload>;
-    fn make_transaction(&self) -> VerifiedTransaction;
+    fn make_transaction(&mut self) -> VerifiedTransaction;
```

Comment:
Would it also make sense to adjust `make_new_payload` to take a `&mut self` + return nothing? I think this will work based on my reading of bench_driver.rs. I think the current setup is a bit confusing + will lead to a lot of copying.

Reply:
sure, made this change you suggested.